server/webpack.config.js

```diff
@@ -0,0 +1,50 @@
+//@ts-check
+
+"use strict";
+
+const path = require("path");
+
+/**
+ * @type {import('webpack').Configuration}
+ */
+const config = {
+    devtool: "source-map",
+    // the entry point of this extension,
+    // 📖 -> https://webpack.js.org/configuration/entry-context/
+    entry: "./src/server.ts",
+    externals: {
+        // the vscode-module is created on-the-fly and must be excluded.
+        // Add other modules that cannot be webpack'ed, 📖 -> https://webpack.js.org/configuration/externals/
+        vscode: "commonjs vscode",
+        "vscode-languageserver": "vscode-languageserver",
+    },
+    module: {
+        rules: [{
+            exclude: /node_modules/,
+            test: /\.(ts|js)$/,
+            use: [{
+                loader: "ts-loader",
+                options: {
+                    compilerOptions: {
+                        "module": "es6", // override `tsconfig.json` so that TypeScript emits native JavaScript modules.
+                    },
+                },
+            }],
+        }],
+    },
+    output: {
+        devtoolModuleFilenameTemplate: "../[resource-path]",
+        filename: "server.js",
+        libraryTarget: "commonjs2",
+        // the bundle is stored in the parent's 'client/server' directory (check package.json),
```

Comment:
LOL I'm finally starting to understand that.

* The `the parent's 'client/server' directory` version sounds extremely weird.
* I'd probably just leave that:

```suggestion
        // the bundle is stored in the “../client/server/” directory (check package.json),
```